core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -558,6 +560,54 @@ private static Optional<TServerInstance> checkServer(ClientContext context, Stri
         .map(address -> new TServerInstance(address, stat.getEphemeralOwner()));
   }
 
+  public static void validate(TabletMetadata tm) {
+    if (!tm.fetchedCols.contains(ColumnType.FILES) || !tm.sawPrevEndRow) {
+      return;
+    }
+
+    Collection<StoredTabletFile> files = tm.getFiles();
+
+    Text prevEndRowText = tm.getPrevEndRow();
```

Comment:
We can simplify this code and use existing code that has lots of unit test already by doing a few things.


Can get a data range for the tablet.

```java
Range tabletRange = tm.getExtent().toDataRange();
```

Then pass the tablet range to `isFileRangeValid()` function and inside the function call `tabletRange.clip(fileRange,true)!=null` to check if a tablet and file do overlap.